modules/electrophysiology_browser/php/models/datasettags.class.inc

```diff
@@ -0,0 +1,612 @@
+                                                : null,
+                    'parent_id'          => $hed_tag
+                                                ? $hed_tag[0]['ParentID']
+                                                : null,
+                    'schema_id'          => $hed_tag
+                                                ? $hed_tag[0]['SchemaID']
+                                                : null,
+                    'name'               => $hed_tag
+                                                ? $hed_tag[0]['Name']
+                                                : '',
+                    'long_mame'          => $hed_tag
+                                                ? $hed_tag[0]['LongName']
+                                                : '',
+                    'description'        => $hed_tag
+                                                ? $hed_tag[0]['Description']
+                                                : '',
+                    'value'              => $tag['TagValue'],
+                    'has_pairing'        => $tag['HasPairing'],
+                    'additional_members' => $tag['AdditionalMembers'],
+                    'rel_id'             => $tag['ID'],
+                    'pair_rel_id'        => $tag['PairRelID']
+                ];
+            },
+            $tags
+        );
+    }
+
+
+    /**
+     * Returns true if at least one element satisfies the function's criteria
+     *
+     * @param array    $array List of elements
+     * @param callable $fn    Function which determines criteria satisfaction
+     *
+     * @return bool
+     */
+    function arrayAny(array $array, callable $fn): bool
+    {
+        foreach ($array as $value) {
+            if ($fn($value)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+
+    /**
+     * Returns index of the Nth member
+     *
+     * @param string $tag_string HED Tag string
+     * @param int    $n          Member index
+     *
+     * @return int
+     */
+    public function getNthTrailingMemberCommaIndex(
+        string $tag_string,
+        int $n
+    ): int {
+        if ($n < 1) {
+            return strlen($tag_string);
+        }
+
+        $members_to_find        = $n;
+        $open_parentheses_count = 0;
+        $comma_index            = 0;
+        $tag_string_split       = str_split($tag_string);
+        for ($i = 0; $i < count($tag_string_split); $i++) {
+            if ($tag_string_split[$i] === '(') {
+                $open_parentheses_count++;
+            } else if ($tag_string_split[$i] === ')') {
+                $open_parentheses_count--;
+            }
+            if ($open_parentheses_count === 0 && $tag_string_split[$i] === ',') {
+                if (--$members_to_find === 0) {
+                    break;
+                }
+            }
+            $comma_index = $i;
+        }
+        return $comma_index + 1;
```

Comment:
I'm not sure of the context of what the format of tag_string is but this seems like the type of function you might want to write unit tests for.

Reply:
Good point. I opened an issue about writing unit tests for this and a few other functions in the near future: #9123.